### ISSUE TITLE
Syntax properties are not forwarded by `define-typed/untyped-identifier`

### ISSUE THREAD

**Issue by jsmaniac** [racket/typed-racket#315]
### What version of Racket are you using?
6.4.0.8 (20160226-15f47ef)

### What program did you run?
```racket
#lang typed/racket

(module t/u racket
  (provide typed/untyped
           (rename-out [typed/untyped-typed original-req-stx]))
  
  (require typed/untyped-utils
           racket/require-syntax
           (for-syntax syntax/strip-context))
  
  (define-require-syntax (typed/untyped-typed stx)
    (displayln 'typed)
    (syntax-case stx ()
      [(_ m s) (replace-context stx #'(submod m s typed))]))
  
  (define-require-syntax (typed/untyped-untyped stx)
    (displayln 'untyped)
    (syntax-case stx ()
      [(_ m s) (replace-context stx #'(submod m s untyped))]))
  
  (define-typed/untyped-identifier typed/untyped
    typed/untyped-typed
    typed/untyped-untyped))

(module m racket
  (module typed typed/racket)
  (module untyped typed/racket/no-check))

(require 't/u)

;; Works:
(require (original-req-stx "." m))
;; Does not work: typed/untyped: not a require sub-form
(require (typed/untyped "." m))
;; But this works:
(define-syntax renamer (make-rename-transformer #'original-req-stx))
(require (renamer "." m))
```

### If you got an error message, please include it here.

```
typed/untyped: not a require sub-form in: (typed/untyped "." m)
```
### What should have happened?

`typed/untyped` should be accepted as a require sub-form.

I suspect that [typed/untyped-renamer](https://github.com/racket/typed-racket/blob/ca46d80189058d612d0ecbae650dba59d8dc7546/typed-racket-more/typed/untyped-utils.rkt#L32) does not forward the syntax properties of the original identifiers.

**Comment by jsmaniac** [comment]
I tried fixing it by returning `(make-rename-tranformer i)` from `typed/untyped-renamer`, but it doesn't work ([write: cannot marshal value that is embedded in compiled code: #<rename-transformer>](https://travis-ci.org/jsmaniac/typed-racket/builds/112851743#L291), because that's returning a transformer procedure, instead of returning a piece of syntax I think).

`make-rename-transformer` seems to be defined in racket's core, so I don't really know how to emulate its behaviour.

**Comment by stamourv** [comment]
Closed by #323
